Composer/packages/client/src/components/Auth/AuthCard.tsx

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import formatMessage from 'format-message';
+import { Fragment, useEffect, useState } from 'react';
+import { ActionButton } from 'office-ui-fabric-react/lib/Button';
+import { Callout } from 'office-ui-fabric-react/lib/Callout';
+import { Persona, PersonaSize } from 'office-ui-fabric-react/lib/Persona';
+import { ILinkStyles, Link } from 'office-ui-fabric-react/lib/Link';
+import { useRecoilValue } from 'recoil';
+import { NeutralColors } from '@uifabric/fluent-theme';
+import { Stack } from 'office-ui-fabric-react/lib/Stack';
+import { FontSizes } from 'office-ui-fabric-react/lib/Styling';
+
+import {
+  currentUserState,
+  isAuthenticatedState,
+  dispatcherState,
+  showAuthDialogState,
+  showTenantDialogState,
+  requiresGraphState,
+} from '../../recoilModel/atoms';
+import { setTenantId } from '../../utils/auth';
+import { zIndices } from '../../utils/zIndices';
+
+import { AuthDialog } from './AuthDialog';
+import { TenantDialog } from './TenantDialog';
+
+const styles = {
+  username: {
+    fontWeight: 600,
+  },
+  email: {
+    display: 'block',
+    fontSize: FontSizes.small,
+  },
+  link: {
+    root: {
+      display: 'block',
+      fontSize: FontSizes.small,
+      marginTop: 5,
+    },
+  },
+  logoutLink: {
+    root: {
+      display: 'block',
+      fontSize: FontSizes.small,
+    },
+  },
+};
+
+export const AuthCard: React.FC = () => {
+  const [authCardVisible, setAuthCardVisible] = useState<boolean>(false);
+  const { refreshLoginStatus, requireUserLogin, logoutUser, setShowAuthDialog, setShowTenantDialog } = useRecoilValue(
+    dispatcherState
+  );
+  const isAuthenticated = useRecoilValue(isAuthenticatedState);
+  const currentUser = useRecoilValue(currentUserState);
+  const showAuthDialog = useRecoilValue(showAuthDialogState);
+  const showTenantDialog = useRecoilValue(showTenantDialogState);
+  const requiresGraph = useRecoilValue(requiresGraphState);
+
+  useEffect(() => {
+    refreshLoginStatus();
+  }, []);
+
+  const logout = () => {
+    logoutUser();
+  };
+
+  const toggleAuthCardVisibility = () => {
+    setAuthCardVisible(!authCardVisible);
+  };
+
+  const switchTenants = () => {
+    setTenantId('');
+    requireUserLogin();
+    toggleAuthCardVisibility();
+  };
+
+  return (
+    <Fragment>
+      {/* this is the icon that appears at the top of the header bar */}
+      <Persona
+        hidePersonaDetails
+        id={'persona'}
```

Comment:
since it's on navbar and it's always showing, this id needs to be unique across the app. can we make sure of that? probably not, so it's better to use a ref for callout target 😄

Reply:
I'm not sure how to achieve this! I'll ping you in teams.